src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java

```diff
@@ -405,18 +412,37 @@ public FormValidation doCheckServer(@QueryParameter String value, @QueryParamete
 
             try {
                 Hashtable<String,String> props = new Hashtable<String,String>();
-                if(managerDN!=null && managerDN.trim().length() > 0  && !"undefined".equals(managerDN)) {
-                    props.put(Context.SECURITY_PRINCIPAL,managerDN);
+                props.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
+                props.put(Context.PROVIDER_URL, LDAPSecurityRealm.toProviderUrl(server,rootDN));
+
+                LdapContext ctx = new InitialLdapContext(props, null);
+                StartTlsResponse tls = (StartTlsResponse) ctx.extendedOperation(new StartTlsRequest());
+                SSLSession session = tls.negotiate();
+
+                if (!session.isValid()) {
+                    throw new IOException("Couldn't negotiate StartTls session, session is invalid");
+                }
+                ctx.addToEnvironment(Context.SECURITY_AUTHENTICATION, "simple");
```

Comment:
is this needed, it would appear to be an unrelated change and would preclude the use of any stronger sasl algorithms?

Reply:
```
                ctx.addToEnvironment(Context.SECURITY_AUTHENTICATION, "simple");
```
is copy pasted from https://docs.oracle.com/javase/jndi/tutorial/ldap/ext/starttls.html#TLS%20with%20Simple%20Authentication

I did see the `External SASL Authentication` section in that link, however I wasn't familiar enough with SASL or the Jenkins UI to add additional form controls.